ops/_tracing/api.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""FIXME docstring."""
+
+from __future__ import annotations
+
+from typing import Any
+
+import ops
+
+from .vendor.charms.certificate_transfer_interface.v1.certificate_transfer import (
+    CertificateTransferRequires,
+)
+from .vendor.charms.tempo_coordinator_k8s.v0.tracing import TracingEndpointRequirer
+
+# NOTE: the shape of certificate_transfer is `{"certificates": list[str]}`
+# it is trivial to read the certificates out manually
+
+
+class Tracing(ops.Object):
+    """FIXME docstring."""
+
+    _certificate_transfer: CertificateTransferRequires | None
+
+    def __init__(
+        self,
+        charm: ops.CharmBase,
+        tracing_relation_name: str,
+        ca_relation_name: str | None = None,
+        ca_data: str | None = None,
+    ):
+        """FIXME docstring.
+
+        Include `ops[tracing]` in your dependencies.
+
+        Declare the relations that the charm supports:
+
+        ```yaml
+        requires:
+            charm-tracing:
+                interface: tracing
+                limit: 1
+            send-ca-cert:
+                interface: certificate_transfer
+                limit: 1
+        ```
+
+        Initialise the tracing thing (FIXME) with the names of these relations.
+
+        ```py
+        import ops.tracing
+
+        class SomeCharm(ops.CharmBase):
+            def __init__(self, framework: ops.Framework):
+                ...
+                self._tracing = ops.tracing.Tracing(
```

Comment:
We want `ops.tracing.Tracing._tracing` (that is a *lot* of "tracing"!) to be private, but in the charm class I think it's fine to be:

```suggestion
                self.tracing = ops.tracing.Tracing(
```

I worry a little that it's not obvious why you need to keep a reference to this object around, but perhaps charmers are used to that from other libraries.

Reply:
That's a lot of tracing, indeed.
Btw., I'm open to some other name for the first-party charm object.

Perhaps `CharmTracing` is better in case the charm in question features some other kinda of `Tracing`, like the workload tracers.

Also, it could be `Tracer` or `CharmTracer` because the instance is a thing, as opposed to the library.

Let's cover that during the standup.